### PR TITLE
monero-wallet-cli: added locked_sweep_all command

### PR DESCRIPTION
**Pull request by jcktm**
This implements locked_sweep_all for the simplewallet and fixes a small bug in sweep_main

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -4840,9 +4849,44 @@ bool simple_wallet::sweep_main(uint64_t below, const std::vector<std::string> &a
...
+    }
+    unlock_block = bc_height + locked_blocks;
+
+    local_args.erase(local_args.begin() + 1);
+  }
+
   std::vector<uint8_t> extra;
   bool payment_id_seen = false;
-  if (2 >= local_args.size())
+  if (2 <= local_args.size())
```
Comment (stoffu) [inline src/simplewallet/simplewallet.cpp]: This seems wrong. The block shouldn't be executed when `local_args.size()<2`.

Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Exactly... this fixes that bug, the block isn't executed when local_args.size() < 2. Perhaps `if (local_args.size() >= 2)` would read better?

Comment (stoffu) [reply src/simplewallet/simplewallet.cpp]: Oops, I read it the wrong way! Yes, I'd prefer `local_args.size() >= 2` much better. Thanks.

Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Fixed, thanks

**Comment by jcktm** [comment]
This references issue #3147

**Comment by stoffu** [comment]
> loki-wallet-cli: added locked_sweep_all command

Please fix the commit message, too.

**Comment by jcktm** [comment]
Fixed

**Comment by luigi1111** [comment]
@jcktm and please rebase. :)

**Comment by moneromooo-monero** [comment]
ping, needs rebasing.

File: src/simplewallet/simplewallet.cpp [hunk 2]
```diff
@@ -5040,9 +5049,44 @@ bool simple_wallet::sweep_main(uint64_t below, const std::vector<std::string> &a
...
+    }
+    unlock_block = bc_height + locked_blocks;
+
+    local_args.erase(local_args.begin() + 1);
+  }
+
   std::vector<uint8_t> extra;
   bool payment_id_seen = false;
-  if (2 >= local_args.size())
+  if (local_args.size() >= 2)
```
Comment (moneromooo-monero) [inline src/simplewallet/simplewallet.cpp]: Why this change ? Was it wrong in the first place ?

File: src/simplewallet/simplewallet.cpp [hunk 3]
```diff
@@ -2169,6 +2169,10 @@ simple_wallet::simple_wallet()
...
+  m_cmd_binder.set_handler("locked_sweep_all",
+                           boost::bind(&simple_wallet::locked_sweep_all, this, _1),
+                           tr("locked_sweep_all [index=<N1>[,<N2>,...]] [<priority>] [<ring_size>] <address> <lockblocks> [<payment_id>]"),
+                           tr("Send all unlocked balance to an address and lock it for <lockblocks> (max. 1000000). If the parameter \"index<N1>[,<N2>,...]\" is specified, the wallet sweeps outputs received by those address indices. If omitted, the wallet randomly chooses an address index to be used. In any case, it tries its best not to combine outputs across multiple addresses. <priority> is the priority of the sweep. The higher the priority, the higher the transaction fee. Valid values in priority order (from lowest to highest) are: unimportant, normal, elevated, priority. If omitted, the default value (see the command \"set priority\") is used. <ring_size> is the number of inputs to include for untraceability."));
```
Comment (stoffu) [inline src/simplewallet/simplewallet.cpp]: This description

> In any case, it tries its best not to combine outputs across multiple addresses.

doesn't seem to apply here; with the sweeping operation, one either specifies the address indices to be swept (so that outputs belonging to different indices get mixed), or omits specifying the address index (so that outputs belonging to one address index get swept).

File: src/simplewallet/simplewallet.cpp (hunk 2 above)
Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Yes

File: src/simplewallet/simplewallet.cpp (hunk 3 above)
Comment (jcktm) [reply src/simplewallet/simplewallet.cpp]: Removed
